pkg/daemon/controller/controller.go

```diff
@@ -138,9 +144,26 @@ func RunController(ctx context.Context, apiServerURL, saCA, saToken string) erro
 }
 
 // NewServiceController returns a pointer to a ServiceController object
-func NewServiceController(ctx context.Context, client client.Client, mgr winsvc.Mgr, nodeName string) *ServiceController {
-	return &ServiceController{client: client, Mgr: mgr, ctx: ctx, nodeName: nodeName,
-		watchNamespace: wmcoNamespace}
+// cmdRunner will be set to the default psCmdRunner if nil
+func NewServiceController(ctx context.Context, client client.Client, mgr winsvc.Mgr, cmdRunner PowershellCommandRunner,
+	nodeName string) *ServiceController {
+	if cmdRunner == nil {
+		cmdRunner = &psCmdRunner{}
+	}
+	return &ServiceController{client: client, Mgr: mgr, ctx: ctx, psCmdRunner: cmdRunner,
+		nodeName: nodeName, watchNamespace: wmcoNamespace}
+}
+
+// psCmdRunner implements the PowershellCommandRunner interface
+type psCmdRunner struct{}
+
+// Run runs the command with the PowerShell on PATH
+func (r *psCmdRunner) Run(cmd string) (string, error) {
```

Comment:
nit: I'd prefer the `psCmdRunner` struct and its method(s) be moved closer to the interface it implements for readability reasons